tests/runner.py

```diff
@@ -196,6 +196,33 @@ def modified(self):
   return decorated
 
 
+@contextlib.contextmanager
+def shared_modify(updates):
+  """A context manager that updates shared."""
```

Comment:
this text is a bit confusing at first glance but I suppose it's just because of the existing naming convention.

Reply:
Sorry, still working on this - I was wrong, this can't work, so I removed it.

When tests are passing again I'll comment with a summary of the new changes.